queries.sql

```diff
@@ -8,3 +8,67 @@ SELECT name, escape_attempts from animals WHERE weight_kg > 10.5;
 SELECT * from animals WHERE neutered = true;
 SELECT * from animals WHERE NOT name = 'Gabumon';
 SELECT * from animals WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+-- update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+
+-- Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
+SELECT * FROM animals;
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+-- Update the animals table by setting the species column to pokemon for all animals that don't have species already set
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+SELECT * FROM animals;
+
+-- delete all records in the animals table, then roll back the transaction.
+-- delete
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+-- ROLLBACK
+ROLLBACK;
+SELECT * FROM animals;
+
+-- Delete all animals born after Jan 1st, 2022
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '01-01-2022';
+SELECT * FROM animals;
+
+-- Create a save point for the transaction
+BEGIN;
+SAVEPOINT savePointOne;
+
+-- Update all animals' weight to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+
+-- Rolling back to the SAVEPOINT
+ROLLBACK To savePointOne;
+SELECT * FROM animals;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+
+-- How many animals have never tried to escape?
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+
+-- Who escapes the most, neutered or not neutered animals?
+SELECT MAX(escape_attempts) FROM animals WHERE neutered = true OR false;
+
```

Comment:
- [ ] The requirement is to get the animal who escaped the most from both types neutered and not neutered. Kindy query that animal who escaped the most from both types. You can use group by to get most escaped animal from both types :+1:

Reply:
done